Libraries/LibWebView/BrowserProcess.cpp

```diff
@@ -114,16 +114,20 @@ void UIProcessConnectionFromClient::die()
     s_connections.remove(client_id());
 }
 
-void UIProcessConnectionFromClient::create_new_tab(Vector<ByteString> urls)
+void UIProcessConnectionFromClient::create_new_tab(Vector<ByteString> urls, CreateNewTab::Resolver resolve)
 {
     if (on_new_tab)
         on_new_tab(sanitize_urls(urls, Application::settings().new_tab_page_url()));
+
+    return resolve();
 }
 
-void UIProcessConnectionFromClient::create_new_window(Vector<ByteString> urls)
+void UIProcessConnectionFromClient::create_new_window(Vector<ByteString> urls, CreateNewWindow::Resolver resolve)
```

Comment:
as the resolver is an object, its variable name should be a noun, not a verb. I checked the surrounding code to see if you had created a new member function called `resolve` when I first glanced at this, which is not good.